bin/offline-vm-setup.sh

```diff
@@ -29,68 +126,66 @@ autoinstall:
+    layout:
+      sizing-policy: all
+      name: lvm
+      match:
+        path: /dev/vda
+        size: largest
   ssh:
-    install-server: yes
+    allow-pw: false
+    install-server: true
+  apt:
+    fallback: offline-install
+  user-data:
+    hostname: ${VM_NAME[i]}
+    users:
+    - default
+    - name: demo
+      groups: sudo
+      sudo: ALL=(ALL) NOPASSWD:ALL
+      shell: /bin/bash
+      ssh_authorized_keys: 
+        - $SSHKEY_HUMAN
+        - $SSHKEY_DEMO
 EOF
 }
 
-create_assethost () {
-    name="$1"
-
-    prepare_config "$name"
-
-    # if you want to run the installation manually remove the `--noautoconsole` flag and the ds= part from `--extra-args`
-    sudo virt-install \
-       --name "$name" \
-        --ram 8192 \
-       --disk path=/var/kvm/images/"$name".img,size=100  \
-       --vcpus 4 \
-       --network bridge=br0 \
-       --graphics none \
-       --osinfo detect=on,require=off \
-       --noautoconsole \
-       --location /home/demo/wire-server-deploy/ubuntu.iso,kernel=casper/vmlinuz,initrd=casper/initrd \
-       --extra-args "console=ttyS0,115200n8 autoinstall ds=nocloud-net;s=http://172.16.0.1:3003/$name"
-}
+create_vm () {
+  prepare_config "${VM_NAME[i]}"
 
-create_node () {
-    name="$1"
-
-    prepare_config "$name"
-
-    # if you want to run the installation manually remove the `--noautoconsole` flag and the ds= part from `--extra-args`
-    sudo virt-install \
-        --name "$name" \
-        --ram 8192 \
-        --disk path=/var/kvm/images/"$name".img,size=100 \
-        --vcpus 6 \
-        --network bridge=br0 \
-        --graphics none \
-        --osinfo detect=on,require=off \
-        --noautoconsole \
-        --location /home/demo/wire-server-deploy/ubuntu.iso,kernel=casper/vmlinuz,initrd=casper/initrd \
-        --extra-args "console=ttyS0,115200n8 autoinstall ds=nocloud-net;s=http://172.16.0.1:3003/$name"
+  sudo virt-install \
+    --name "${VM_NAME[i]}" \
+    --ram 8192 \
+    --disk path=/var/lib/libvirt/images/"${VM_NAME[i]}".qcow2,size=100 \
+    --vcpus 4 \
+    --network bridge=virbr0 \
+    --graphics none \
+    --osinfo detect=on,require=off \
+    --noautoconsole \
+    --location "$DEPLOY_DIR"/ubuntu.iso,kernel=casper/vmlinuz,initrd=casper/initrd \
+    --extra-args "console=ttyS0,115200n8 autoinstall ds=nocloud-net;s=http://192.168.122.1:3003/${VM_NAME[i]}"
 }
 
-if [ "$1" = "serve_nocloud" ]; then
-    mkdir -p "$nocloud_basedir"
-    cd "$nocloud_basedir"
-    python3 -m http.server 3003
-fi
-
-if [ "$1" = "create_node" ]; then
+for (( i=0; i<${#VM_NAME[@]}; i++ )); do
+  if sudo virsh list --all | grep -Fq "${VM_NAME[i]}"; then
+    msg ""
+    msg "ATTENTION - VM ""${VM_NAME[i]}"" already exists"
+    msg ""
+    continue
+  else
     set -u
-    name="$2"
-    create_node "$name"
-fi
-
-if [ "$1" = "create_assethost" ]; then
-    set -u
-    name="$2"
-    create_assethost "$name"
-fi
+    msg "Creating VM ""${VM_NAME[i]}"" with IP ""${VM_IP[i]}"" ..."
```

Comment:
If we have to set this up again, we're going to have a LOT of duplicate entries in /etc/hosts